### PR TITLE
Add TLS support

### PR DESCRIPTION
**Pull request by Vampouille**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,8 +1,12 @@
...
-    - "6432:6432"
-  volumes:
-    - ./etc/pgbouncer:/etc/pgbouncer:ro
-    - ./log/postgresql:/var/log/postgresql:rw
-  restart: always
+version: '2'
+services:
+  pgbouncer:
+    build: .
+    image: camptocamp/pgbouncer
```
Comment (raphink) [inline docker-compose.yml:5]: build + image is not possible

File: docker-compose.yml [hunk 2]
```diff
@@ -1,8 +1,12 @@
...
+    ports:
+      - "6432:6432"
+    volumes:
```
Comment (raphink) [inline docker-compose.yml:8]: Please keep the local volumes, they're used for testing

File: entrypoint.sh [hunk 3]
```diff
@@ -1,13 +1,23 @@
...
 set -e
 
-PG_LOG=/var/log/postgresql/
 PG_CONFIG=/etc/pgbouncer/pgbouncer.ini
 PG_USER=postgres
 
-mkdir -p ${PG_LOG}
-chmod -R 755 ${PG_LOG}
-chown -R ${PG_USER}:${PG_USER} ${PG_LOG}
+# Generate key and certificate for ssl connection
```
Comment (raphink) [inline entrypoint.sh:7]: Why remove the PG_LOG creation + chmod?

File: entrypoint.sh [hunk 4]
```diff
@@ -1,13 +1,23 @@
...
+if [ ! -r /etc/pgbouncer/ssl/server.key ]; then
+	echo "Generating new key and certificate for SSL connection"
+	CWD=`pwd`
+	cd /etc/pgbouncer/ssl/
+	openssl genrsa -des3 -passout pass:xxxx -out server.pass.key 2048
+	openssl rsa -passin pass:xxxx -in server.pass.key -out server.key
+	rm -f server.pass.key
+	openssl req -new -key server.key -out server.csr -subj "/C=FR/O=CampToCamp"
```
Comment (raphink) [inline entrypoint.sh]: O=CampToCamp? Maybe provide an env variable for this?

File: docker-compose.yml (hunk 1 above)
Comment (Vampouille) [reply docker-compose.yml:5]: did you test it ?

File: entrypoint.sh (hunk 3 above)
Comment (Vampouille) [reply entrypoint.sh:7]: For docker images, log should only be sent to stdout, creating log file in container is useless and impact performance by writing on aufs layer.

File: docker-compose.yml (hunk 1 above)
Comment (Vampouille) [reply docker-compose.yml:5]: It works on my workstation

File: docker-compose.yml (hunk 2 above)
Comment (Vampouille) [reply docker-compose.yml:8]: Done with https://github.com/camptocamp/docker-pgbouncer/commit/7c34743d3d7e28e1ce6cae8eaaea4c97e30c26f1

**Pull request merged**